lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1143,6 +1153,18 @@ void TypeChecker::checkIgnoredExpr(Expr *E) {
     return;
   }
 
+  bool suggestDiscardResult = false;
+  SWIFT_DEFER {
+    if (suggestDiscardResult) {
+      diagnose(valueE->getLoc(), diag::discard_result_of_call)
+      .fixItInsert(valueE->getStartLoc(), "_ = ");
+    }
```

Comment:
Please don't defer side effects like this, either. That's a bit too subtle for C++ code.